### PR TITLE
LibJS: Implement `BigInt.asIntN` and `BigInt.asUintN`

### PR DESCRIPTION
**Pull request by trflynn89**
```
test/built-ins/BigInt                            74/74    (100.00%) [ ✅ 74 ]
test/built-ins/BigInt/asIntN                     14/14    (100.00%) [ ✅ 14 ]
test/built-ins/BigInt/asUintN                    14/14    (100.00%) [ ✅ 14 ]
test/built-ins/BigInt/parseInt                    1/1     (100.00%) [ ✅ 1  ]
test/built-ins/BigInt/prototype                  24/24    (100.00%) [ ✅ 24 ]
test/built-ins/BigInt/prototype/toLocaleString    1/1     (100.00%) [ ✅ 1  ]
test/built-ins/BigInt/prototype/toString         11/11    (100.00%) [ ✅ 11 ]
test/built-ins/BigInt/prototype/valueOf           8/8     (100.00%) [ ✅ 8  ]
```

### CODE REVIEW

**Comment by trflynn89** [comment]
The first commit is of course completely unrelated, but for some reason, CI started hitting this ubsan error there:
```
CRASHED test-js/test-js (315.177896s)
         Test:   test-js (crashed)
UBSAN: overflow when casting from 'double' to 'unsigned short'
UBSAN: at /home/vsts/work/1/s/Userland/Libraries/LibJS/Runtime/Date.cpp, line 70, column: 29
UB is configured to be deadly
```

**Comment by trflynn89** [comment]
For posterity because the last comment won't make complete sense anymore, as the commit it refers to is gone: The commit was just this diff:
```diff
diff --git a/Userland/Libraries/LibJS/Runtime/AbstractOperations.h b/Userland/Libraries/LibJS/Runtime/AbstractOperations.h
index 46ce41edb0..db1fc4fa8f 100644
--- a/Userland/Libraries/LibJS/Runtime/AbstractOperations.h
+++ b/Userland/Libraries/LibJS/Runtime/AbstractOperations.h
@@ -146,8 +146,12 @@ auto modulo(T x, U y) requires(IsArithmetic<T>, IsArithmetic<U>)
 
 auto modulo(Crypto::BigInteger auto const& x, Crypto::BigInteger auto const& y)
 {
-    VERIFY(y != "0"_bigint);
+    static const auto zero = "0"_bigint;
+    VERIFY(y != zero);
+
     auto result = x.divided_by(y).remainder;
+    if (result.is_negative() && (result.unsigned_value() == zero))
+        result.negate();
     if (result.is_negative())
         result = result.plus(y);
     return result;
```

The issue was indeed some BigInt negative zeros being converted to a double then an unsigned int, which is undefined behavior. The first commit here now makes it an invariant that SignedBigInteger cannot be negative zero.

**Comment by trflynn89** [comment]
This also comes with a side of fixing some temporal tests :)
```
Before: test/built-ins/Temporal    [ ✅ 3298  ❌ 109   💥️ 15    📝 83 ] 
After:  test/built-ins/Temporal    [ ✅ 3302  ❌ 105   💥️ 15    📝 83 ] 
```

File: Userland/Libraries/LibJS/Tests/builtins/BigInt/BigInt.asIntN.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        expect(BigInt.asIntN(4, -3n)).toBe(-3n);
+        expect(BigInt.asIntN(4, -2n)).toBe(-2n);
+        expect(BigInt.asIntN(4, -1n)).toBe(-1n);
+        expect(BigInt.asIntN(4, 0n)).toBe(0n);
+        expect(BigInt.asIntN(4, 1n)).toBe(1n);
+        expect(BigInt.asIntN(4, 2n)).toBe(2n);
+        expect(BigInt.asIntN(4, 3n)).toBe(3n);
+
+        const extremelyBigInt = 123456789123456789123456789123456789123456789123456789n;
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/BigInt/BigInt.asIntN.js:63]: :yakkiss:

**Review by linusg — APPROVED** [review]

**Pull request merged**